attachment_s3/models/ir_attachment.py

```diff
@@ -55,7 +55,7 @@ def _get_s3_bucket(self, name=None):
         region_name = os.environ.get('AWS_REGION')
         access_key = os.environ.get('AWS_ACCESS_KEY_ID')
         secret_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
-        bucket_name = name or os.environ.get('AWS_BUCKETNAME')
+        bucket_name = name or os.environ.get("AWS_BUCKETNAME") + "-" + self.env.cr.dbname
```

Comment:
This change implies that `bucket_name` is always set.

This breaks the condition at line 68.

Reply:
Maybe this feature can be added using another env variable, to avoid breaking old instances

For example:

```suggestion
        bucket_name = name or os.environ.get("AWS_BUCKETNAME")
        if bucket_name and os.environ.get("AWS_BUCKETNAME_MULTI_DB"):
            bucket_name = "%s-%s" % (bucket_name, self.env.cr.dbname)
```

Just a suggestion

Reply:
@ivantodorovich This would indeed be safer to avoid an angry mob of single database users.

Reply:
@ivantodorovich @yvaucher What about a migration script to rename the existing bucket?

Reply:
S3 buckets cannot be renamed, so the only option would be to create a new one and migrate the files from the old one. And even if that was part of the migration, it would be unpleasant for Odoo administrators who don't have control over the name of the bucket.

Could the bucket name be an `ir.config_parameter`? Then you could define a bucket naming pattern in an environment variable, ie. `AWS_BUCKETNAME_TEMPLATE=odoo-filestore-{db}` and that would set the `ir.config_parameter` called `ir_attachment.s3_bucket` to `odoo-filestore-mydatabase` in a database called `mydatabase`. This would be only a template for new bucket names and wouldn't affect the existing databases & buckets in any way.

Reply:
@mlaitinen indeed with an ir.config_parameter it would make it even more configurable on a per database basis. This might no be necessary for the current need.

Detecting the presence of `{db}` patern in AWS_BUCKET_NAME and replace it on the fly would work.
...I doubt anyone has a `{db}` string in their bucket names. (I doubt it's even possible, if it is... it's an acceptable limitation)


Side note: Probably not what we are looking for but for setting an ir.config_parameter (not with a pattern) from an environment variable we already have this module:

https://github.com/OCA/server-env/tree/14.0/server_environment_ir_config_parameter

Reply:
Following @mlaitinen suggestion:

```suggestion
        bucket_name = name or os.environ.get("AWS_BUCKETNAME")
        # replaces {db} by the database name to handle multitenancy
        bucket_name.format(db=self.env.cr.dbname)
```

Reply:
@max3903 with this last option you would have to append `-{db}` to your AWS_BUCKETNAME env var.
Seems simpler to me than to migrate hundreds of files from a bucket to an other :wink:

Reply:
@yvaucher If we can support this without any code change, even better... :+1:

Reply:
Does this apply to swift storage as well?

Reply:
@dannyadair this only applies to `attachment_s3` module.
We could do a similar change for SWIFT.

If you are willing you can try to apply the same change as #218 to `attachment_swift` module.

Reply:
Thanks @yvaucher sorry that took a moment. Because of the way attachment_swift works - setting the full name incl. container at write time - this was actually just a single line change and some documentation. https://github.com/camptocamp/odoo-cloud-platform/pull/256